src/lists/getList.test.ts

```diff
@@ -0,0 +1,181 @@
+import axios, { AxiosError, AxiosResponse } from "axios";
+import { getList } from "./getList";
+import { ListNotFoundError } from "../errors/ListNotFoundError";
+import {
+  EmailOctopusError,
+  ApiKeyInvalidError,
+  InvalidParametersError,
+  NotFoundError,
+  UnauthorisedError,
+} from "../emailOctopus";
+import { UnknownError } from "../errors/UnknownError";
+
+jest.mock("axios", () => {
```

Comment:
Could we use the `AxiosMockAdapter` here, please? This will enable us to do better mocking in general and test errors much easier. You can see an example here: https://github.com/kartoffelkraft/email-octopus-ts/blob/main/src/lists/createContact.test.ts

It also improves type-safety :) 

```tsx
const mock = new AxiosMockAdapter(axios);
// ...

it("should return the list", async () => {
  const expectedList = {
    id: listId,
    name: "List 1"
  };

  mock.onGet().replyOnce(200, expectedList);

  const result = await getList(FAKE_API_KEY)({ listId });

  expect(axios.get).toBeCalledWith(
    `https://emailoctopus.com/api/1.6/lists/${listId}`,
    { params: { api_key: FAKE_API_KEY } }
  );
  expect(result).toEqual(expectedList);
});
```

Reply:
Hello @igeligel , I made the changes as requested, I adjusted the mocks to use the AxiosMockAdapter